auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/UmsClient.java

```diff
@@ -1051,17 +1051,34 @@ public String getIdentityProviderMetadataXml(String requestId, String accountId)
         return wags;
     }
 
-    private UserManagementProto.Assignee getAssignee(String userCrn) {
-        String accountId = Crn.fromString(userCrn).getAccountId();
+    private UserManagementProto.Assignee getAssignee(String assigneeCrn) {
+        checkNotNull(assigneeCrn);
+        Crn crn = Crn.safeFromString(assigneeCrn);
+        String accountId = crn.getAccountId();
         validateAccountIdWithWarning(accountId);
-        UserManagementProto.Assignee.Builder assignee = UserManagementProto.Assignee.newBuilder()
-                .setAccountId(accountId);
-        if (Crn.isCrn(userCrn) && Crn.ResourceType.MACHINE_USER.equals(Crn.fromString(userCrn).getResourceType())) {
-            assignee.setMachineUserNameOrCrn(userCrn);
-        } else {
-            assignee.setUserIdOrCrn(userCrn);
+        UserManagementProto.Assignee roleAssignee;
+
+        UserManagementProto.Assignee.Builder assignee = UserManagementProto.Assignee.newBuilder().setAccountId(accountId);
+        switch (crn.getResourceType()) {
+            case USER:
+                assignee.setUserIdOrCrn(assigneeCrn);
+                roleAssignee = assignee.build();
```

Comment:
why do we need `roleAssignee` variable? we can return with `assignee.build();` after switch case as before I think

Reply:
This is used in the case related logs, for example:
```LOGGER.info("Found user '{}' for assignee Crn: '{}'.", roleAssignee.getUserIdOrCrn(), assigneeCrn);```
or
```LOGGER.info("Found group '{}' for assignee Crn: '{}'.", roleAssignee.getGroupNameOrCrn(), assigneeCrn);```
So for the return variable is not needed to build the assignee as `roleAssignee`. However for logging purposes this is needed, because of we can get `roleAssignee.getUserIdOrCrn()`, `roleAssignee.getMachineUserNameOrCrn()` and `roleAssignee.getGroupNameOrCrn()` on this way based on the case `USER`, `MACHINE_USER` or `GROUP`.

Reply:
for logging you can use the assigneeCrn, which was set before

Reply:
but, i am not sure i get the logging itself, it logs the same crn twice, you can just log the assigneeCrn once

Reply:
so now you set the assigneeCrn and you log it twice, once from roleAssignee (which was set with assigneeCrn), once the assigneeCrn itself